### PR TITLE
feat: direct connect associations and attachments

### PR DESCRIPTION
**Pull request by James-Quigley**
https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/directconnect@v1.4.1/types#DirectConnectGatewayAssociation

https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/directconnect@v1.4.1/types#DirectConnectGatewayAttachment

### CODE REVIEW

File: resources/directconnect_gateways.go [hunk 1]
```diff
@@ -62,6 +64,124 @@ func DirectconnectGateways() *schema.Table {
...
+						Description: "The ID of the AWS account that owns the virtual private gateway.",
+						Type:        schema.TypeString,
+					},
+				},
+			},
+			{
+				Name:        "aws_directconnect_gateway_attachments",
+				Description: "Information about the attachment between a Direct Connect gateway and virtual interfaces.",
+				Resolver:    fetchDirectconnectGatewayAttachments,
+				Columns: []schema.Column{
```
Comment (roneli) [inline resources/directconnect_gateways.go:145]: I would also save the original DirectConnectGatewayId from AWS as resource_id (hopefully soon we will update the sdk to allow users to define any column as the PK)

**Review by roneli — CHANGES_REQUESTED** [review]
Looks great, a few minor changes and remarks

File: resources/directconnect_gateways.go [hunk 2]
```diff
@@ -62,6 +64,124 @@ func DirectconnectGateways() *schema.Table {
...
+					{
+						Name:        "virtual_gateway_id",
+						Description: "The ID of the virtual private gateway. Applies only to private virtual interfaces.",
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "virtual_gateway_owner_account",
+						Description: "The ID of the AWS account that owns the virtual private gateway.",
+						Type:        schema.TypeString,
+					},
```
[the remark this answers is not in the document]
Comment (James-Quigley) [reply resources/directconnect_gateways.go:138]: its deprecated. Should I still include it?

[the remark this answers is not in the document]
Comment (roneli) [reply resources/directconnect_gateways.go:144]: No need my bad

**Review by roneli — APPROVED** [review]
Looks good to me, generate the docs for the new tables (run docs.go inside docs directory) and we can merge

**Comment by James-Quigley** [comment]
> Looks good to me, generate the docs for the new tables (run docs.go inside docs directory) and we can merge

**Pull request merged**